modules/state_machine/states/wall.gd

```diff
@@ -0,0 +1,50 @@
+extends "res://modules/state_machine/states/state.gd"
+
+var normal = Vector2(0,0)
+var jump_length = 500
+signal wall_finished
+signal wall_jumping
+signal is_on_wall
```

Comment:
Should really reconsider all these signals, maybe they should be refactored

Reply:
is_on_wall sounds like a function wich return a boolean, not a signal. Dont know if change it's use, but at least you should rename it to enter_wall, exit_wall and jumped_from_wall or something like that.

Reply:
This makes lots of sense, but I'm really reconsidering just figuring out another way to trigger these, any suggestions? They are only used for animations afak.

Reply:
Emitting signals give more flexibility, you can connect animation plays with different animation names according yo your character scene, or also you an spawn particles.